pkg/private/tar/tar_writer.py

```diff
@@ -129,7 +129,7 @@ def _addfile(self, info, fileobj=None):
       # Enforce the ending / for directories so we correctly deduplicate.
       if not info.name.endswith('/'):
         info.name += '/'
-    if not self.allow_dups_from_deps and self._have_added(info.name):
+    elif not self.allow_dups_from_deps and self._have_added(info.name):
```

Comment:
Please add a comment here explaining why it needs to be this way. It may help prevent future contributors from changing the logic inadvertently.